bundles/org.openhab.binding.solarforecast/src/main/java/org/openhab/binding/solarforecast/internal/forecastsolar/handler/ForecastSolarBridgeHandler.java

```diff
@@ -0,0 +1,150 @@
+import java.util.Optional;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.solarforecast.internal.SolarForecastBindingConstants;
+import org.openhab.binding.solarforecast.internal.actions.SolarForecast;
+import org.openhab.binding.solarforecast.internal.actions.SolarForecastActions;
+import org.openhab.binding.solarforecast.internal.actions.SolarForecastProvider;
+import org.openhab.binding.solarforecast.internal.forecastsolar.ForecastSolarObject;
+import org.openhab.binding.solarforecast.internal.forecastsolar.config.ForecastSolarBridgeConfiguration;
+import org.openhab.binding.solarforecast.internal.utils.Utils;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.library.types.PointType;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+
+/**
+ * The {@link ForecastSolarBridgeHandler} is a non active handler instance. It will be triggerer by the bridge.
+ *
+ * @author Bernd Weymann - Initial contribution
+ */
+@NonNullByDefault
+public class ForecastSolarBridgeHandler extends BaseBridgeHandler implements SolarForecastProvider {
+    private final PointType homeLocation;
+
+    private List<ForecastSolarPlaneHandler> parts = new ArrayList<ForecastSolarPlaneHandler>();
+    private Optional<ForecastSolarBridgeConfiguration> configuration = Optional.empty();
+    private Optional<ScheduledFuture<?>> refreshJob = Optional.empty();
+
+    public ForecastSolarBridgeHandler(Bridge bridge, PointType location) {
+        super(bridge);
+        homeLocation = location;
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singleton(SolarForecastActions.class);
+    }
+
+    @Override
+    public void initialize() {
+        ForecastSolarBridgeConfiguration config = getConfigAs(ForecastSolarBridgeConfiguration.class);
+        if (config.location.equals(SolarForecastBindingConstants.AUTODETECT)) {
+            Configuration editConfig = editConfiguration();
+            editConfig.put("location", homeLocation.toString());
+            updateConfiguration(editConfig);
+            config = getConfigAs(ForecastSolarBridgeConfiguration.class);
+        }
+        configuration = Optional.of(config);
+        updateStatus(ThingStatus.ONLINE);
+        refreshJob = Optional.of(
+                scheduler.scheduleWithFixedDelay(this::getData, 0, config.channelRefreshInterval, TimeUnit.MINUTES));
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    /**
+     * Get data for all planes. Synchronized to protect parts map from being modified during update
```

Comment:
`getData` called from refreshJob, `addPlane` / `removePlane` (new / delete Thing) and `getSolarForecasts` (rule access) might occur in parallel  
So access to `parts` list is synchronized

Reply:
> However, since each of the methods you mention are synchronized individually, you are not really protecting `planes` of concurrent write access. For example, `addPlane` and `removePlane` could be called at the same time.

What? Synchronized methods cannot be entered by different threads. Basic data structures like [java.util.Vector](https://docs.oracle.com/javase/8/docs/api/java/util/Vector.html) are using synchronized methods too to prevent parallel / concurrent acces